### PR TITLE
Remove ordering constraint for Givens rotations

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

**Comment by simonbyrne** [comment]
cc @andreasnoack

File: base/linalg/givens.jl [hunk 1]
```diff
@@ -230,14 +230,18 @@ cosine and sine of the rotation angle can be extracted from the `Givens` type wi
 and `G.s` respectively. The arguments `f` and `g` can be either `Float32`, `Float64`,
 `Complex{Float32}`, or `Complex{Float64}`. The `Givens` type supports left multiplication
 `G*A` and conjugated transpose right multiplication `A*G'`. The type doesn't have a `size`
-and can therefore be multiplied with matrices of arbitrary size as long as `i2<=size(A,2)`
-for `G*A` or `i2<=size(A,1)` for `A*G'`.
+and can therefore be multiplied with matrices of arbitrary size as long as
+`max(i1,i2)<=size(A,2)` for `G*A` or `max(i1,i2)<=size(A,1)` for `A*G'`.
```
Comment (tkelman) [inline base/linalg/givens.jl]: run genstdlib

File: base/linalg/givens.jl [hunk 2]
```diff
@@ -230,14 +230,18 @@ cosine and sine of the rotation angle can be extracted from the `Givens` type wi
...
 """
 function givens{T}(f::T, g::T, i1::Integer, i2::Integer)
-    if i1 >= i2
-        throw(ArgumentError("second index must be larger than the first"))
+    if i1 == i2
+        throw(ArgumentError("Indices must be distinct."))
     end
     c, s, r = givensAlgorithm(f, g)
+    if i1 > i2
+        s = -s
```
Comment (andreasnoack) [inline base/linalg/givens.jl]: Shouldn't this have a conjugation as well?

Comment (simonbyrne) [inline base/linalg/givens.jl]: Why would it?

File: base/linalg/givens.jl [hunk 3]
```diff
@@ -250,16 +254,11 @@ the rotation angle can be extracted from the `Givens` type with `G.c` and `G.s`
...
-    if i1 >= i2
-        throw(ArgumentError("second index must be larger than the first"))
-    end
-    c, s, r = givensAlgorithm(A[i1,col], A[i2,col])
-    Givens(i1, i2, convert(T, c), convert(T, s)), r
-end
+givens{T}(A::AbstractMatrix{T}, i1::Integer, i2::Integer, col::Integer) =
+    givens(A[i1,col], A[i2,col],i1,i2)
 
 getindex(G::Givens, i::Integer, j::Integer) = i == j ? (i == G.i1 || i == G.i2 ? G.c : one(G.c)) : (i == G.i1 && j == G.i2 ? G.s : (i == G.i2 && j == G.i1 ? -G.s : zero(G.s)))
```
Comment (simonbyrne) [inline base/linalg/givens.jl]: @andreasnoack see here.

Comment (andreasnoack) [inline base/linalg/givens.jl]: I think this is an error. See line 274. The author must have been sloppy.

Comment (KristofferC) [inline base/linalg/givens.jl]: But he/she sure managed to get it all on one line! :)

**Comment by simonbyrne** [comment]
This adds the fix to `getindex`

**Comment by andreasnoack** [comment]
Thanks

**Pull request merged**